src/openrct2/drawing/Drawing.String.cpp

```diff
@@ -723,7 +729,29 @@ static const utf8 *ttf_process_glyph_run(rct_drawpixelinfo *dpi, const utf8 *tex
 
 static void ttf_process_string(rct_drawpixelinfo *dpi, const utf8 *text, text_draw_info *info)
 {
-    const utf8 *ch = text;
+    UErrorCode err = (UErrorCode)0;
+
+    UnicodeString ustr = UnicodeString::fromUTF8(StringPiece(text));
```

Comment:
These operations are rather expensive to do. I'd rather we didn't do them at every tick... Can we cache this, somehow? (Indeed, we probably shouldn't do them at all for non-LTR languages, but I assume you had that planned already.)